numba/core/typing/builtins.py

```diff
@@ -654,6 +654,26 @@ def generic(self, args, kws):
             sig = signature(ret, *args)
             return sig
 
+@infer
+class StaticGetItemClass(AbstractTemplate):
+    """This handles the "static_getitem" that occurs when a Numba type has a
+    __getitem__ made on it in input source e.g:
+    var = typed.List.empty_list(float64[::1, :])
+    it only allows this on simple numerical types i.e. compound types, like
+    records, are not supported.
+    """
+    key = "static_getitem"
+
+    def generic(self, args, kws):
+        clazz, idx = args
+        accept = (types.NumberClass, types.Boolean)
+        if not isinstance(clazz, accept):
+            return
+        ret = clazz.dtype[idx]
```

Comment:
TBC: I'm not yet sure how this would work if `clazz` is a `Boolean`, as that doesn't appear to have a `dtype`.

Reply:
I think your assertion is correct, use of `types.Boolean` is incorrect/superfluous, as `clazz` would be an instance of a `types.functions.NumberClass` and "boolean" types would work through that.

Reply:
OK - does that just need removing then?

Reply:
Yes, I think so, 3c5da76 contains this change.